tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.cc

```diff
@@ -78,6 +78,52 @@ void TrtShapeOptimizationProfile::ImplicitBatchModeCompatibleStrategy(
   }
 }
 
+// Create a new Dims vector where the dimensions are the result of a binary
+// operation between dimensions in lhs and rhs.
+template <typename BinaryOperation>
+Status ShapeProfileBinaryOp(const std::vector<nvinfer1::Dims>& lhs,
+                            const std::vector<nvinfer1::Dims>& rhs,
+                            BinaryOperation op,
+                            std::vector<nvinfer1::Dims>* out) {
+  std::vector<nvinfer1::Dims> x(lhs.size());
+  if (lhs.size() != rhs.size())
+    return errors::InvalidArgument(
+        "Number of input tensors differ during profile creation");
+  for (int i = 0; i < lhs.size(); i++) {
+    if (lhs[i].nbDims != rhs[i].nbDims)
+      return errors::InvalidArgument(
+          "Number of input dimensions differ during profile creation");
+    x[i].nbDims = lhs[i].nbDims;
+    for (int j = 0; j < lhs[i].nbDims; j++) {
+      x[i].d[j] = op(lhs[i].d[j], rhs[i].d[j]);
+    }
+  }
+  *out = x;
+  return Status::OK();
+}
+
+Status TrtShapeOptimizationProfile::RangeStrategy(
+    const std::vector<std::vector<nvinfer1::Dims>>& collected_shapes) {
+  if (collected_shapes.empty()) return Status::OK();
+
+  std::vector<nvinfer1::Dims> min = collected_shapes[0];
+  std::vector<nvinfer1::Dims> max = min;
+
+  for (int i = 1; i < collected_shapes.size(); i++) {
+    TF_RETURN_IF_ERROR(ShapeProfileBinaryOp(
+        min, collected_shapes[i], [](int a, int b) { return std::min(a, b); },
+        &min));
```

Comment:
Can we replace the lambda with std::min?

Reply:
Not simply, it is overleaded and the lambda is used to resolve ambiguity. Alternatively, we could [cast it to a function pointer](https://stackoverflow.com/a/47109361/9018949) if you prefer that solution.